src/Rule/Callback.php

```diff
@@ -19,12 +23,21 @@ public function __construct(callable $callback)
 
     protected function validateValue($value, DataSetInterface $dataSet = null): Result
     {
-        $result = new Result();
         $callback = $this->callback;
-        /**
-         * @var $callbackResult Result
-         */
         $callbackResult = $callback($value, $dataSet);
+
+        if (!$callbackResult instanceof Result) {
+            throw new CallbackRuleException(
+                sprintf(
+                    'Return value of callback must be an instance of %s, %s returned.',
+                    Result::class,
+                    gettype($callbackResult)
```

Comment:
We need to add extra information about object class if `$callbackResult` is object but not implements `Result`
Else we will have the following messages:

> Return value of callback must be an instance of ..., **"object"** returned.

```suggestion
                    is_object($callbackResult) ? get_class($callbackResult) : gettype($callbackResult)
```